source/client.c

```diff
@@ -2737,58 +2489,54 @@ static enum aws_mqtt_client_request_state s_unsubscribe_send(
                 goto handle_error;
             }
         }
-
-        task_arg->is_local = topic ? topic->is_local : false;
     }
 
-    if (!task_arg->is_local) {
-        if (task_arg->unsubscribe.fixed_header.packet_type == 0) {
-            /* If unsubscribe packet is uninitialized, init it */
-            if (aws_mqtt_packet_unsubscribe_init(&task_arg->unsubscribe, task_arg->connection->allocator, packet_id)) {
-                goto handle_error;
-            }
-            if (aws_mqtt_packet_unsubscribe_add_topic(&task_arg->unsubscribe, task_arg->filter)) {
-                goto handle_error;
-            }
-        }
-
-        message = mqtt_get_message_for_packet(task_arg->connection, &task_arg->unsubscribe.fixed_header);
-        if (!message) {
+    if (task_arg->unsubscribe.fixed_header.packet_type == 0) {
+        /* If unsubscribe packet is uninitialized, init it */
+        if (aws_mqtt_packet_unsubscribe_init(&task_arg->unsubscribe, task_arg->connection->allocator, packet_id)) {
             goto handle_error;
         }
-
-        if (aws_mqtt_packet_unsubscribe_encode(&message->message_data, &task_arg->unsubscribe)) {
+        if (aws_mqtt_packet_unsubscribe_add_topic(&task_arg->unsubscribe, task_arg->filter)) {
             goto handle_error;
         }
+    }
 
-        if (aws_channel_slot_send_message(task_arg->connection->slot, message, AWS_CHANNEL_DIR_WRITE)) {
-            goto handle_error;
-        }
+    message = mqtt_get_message_for_packet(task_arg->connection, &task_arg->unsubscribe.fixed_header);
+    if (!message) {
+        goto handle_error;
+    }
 
-        /* TODO: timing should start from the message written into the socket, which is aws_io_message->on_completion
-         * invoked, but there are bugs in the websocket handler (and maybe also the h1 handler?) where we don't properly
-         * fire the on_completion callbacks. */
-        struct request_timeout_task_arg *timeout_task_arg = s_schedule_timeout_task(task_arg->connection, packet_id);
-        if (!timeout_task_arg) {
-            return AWS_MQTT_CLIENT_REQUEST_ERROR;
-        }
+    if (aws_mqtt_packet_unsubscribe_encode(&message->message_data, &task_arg->unsubscribe)) {
+        goto handle_error;
+    }
 
-        /*
-         * Set up mutual references between the operation task args and the timeout task args.  Whoever runs first
-         * "wins", does its logic, and then breaks the connection between the two.
-         */
-        task_arg->timeout_wrapper.timeout_task_arg = timeout_task_arg;
-        timeout_task_arg->task_arg_wrapper = &task_arg->timeout_wrapper;
+    if (aws_channel_slot_send_message(task_arg->connection->slot, message, AWS_CHANNEL_DIR_WRITE)) {
+        goto handle_error;
     }
 
+    /* TODO: timing should start from the message written into the socket, which is aws_io_message->on_completion
+     * invoked, but there are bugs in the websocket handler (and maybe also the h1 handler?) where we don't properly
```

Comment:
I don't immediately see a ticket related to this. Should we open one if this is a bug?

Reply:
It's just an indentation change and I have no desire to monkey with it.  But we did fix websocket completion callbacks during the mqtt5 implementation